eng/scripts/Deploy-TestResources.ps1

```diff
@@ -19,11 +21,42 @@ function New-StringHash([string[]]$strings) {
     return [BitConverter]::ToString($hashBytes) -replace '-', ''
 }
 
-$testResourcesDirectory = Resolve-Path -Path  "$RepoRoot/areas/$($Area.ToLower())/tests" -ErrorAction SilentlyContinue
+switch ($PSCmdlet.ParameterSetName) {
+    'ByPath' {
+        if(!(Test-Path -Path $Path)) {
+            Write-Error "Path '$Path' does not exist."
+            exit 1
+        }
+    }
+    'ByTool' {
+        $tools = Get-ChildItem -Path "$RepoRoot/tools" -Directory
+
+        $match = $tools | Where-Object { $_.Name -ceq $Tool }
+        if ($match) {
+            $Path = $match.FullName
+        } else {
+            $matchingTools = $tools | Where-Object { $_.Name -like "*$Tool*" }
+            if ($matchingTools.Count -eq 1) {
+                $match = $matchingTools[0]
+                Write-Host "Found tool '$($match.Name)'."
+                $Path = $match.FullName
+            } elseif ($matchingTools.Count -gt 1) {
```

Comment:
What is the magic to get all? `.Mcp`

Reply:
There is no magic.  This currently requires the `-tool` substring match to find only 1 match.  Should I allow it to find and and let the user pass the substring?  e.g.

```ps
$matchingTools = $toolDirectories | where { $_.Name -eq $Tool }
if(!$matchingTools)
  $matchingTools = $toolDirectories | where { $_.Name -like $Tool }
if(!$matchingTools)
  $matchingTools = $toolDirectories | where { $_.Name -like "*$Tool*" }
```

Try to find an exact tool match
Then try the $tool string as a wildcard match
If that still doesn't resolve to anything, try a substring match

Reply:
We should have some way to deploy multiple or even all if needed.

Reply:
I could also simplify that and have the users handle substring/wildcard themselves

`Deploy-TestResources.ps1 -Tool '*'`   or `Deploy-TestResources.ps1 -Tool '*keyvault'`

Reply:
That should work I just want to make sure we have a way to do it and we should also document it.